### DISCUSSION TITLE
Search component id in builder

### DISCUSSION THREAD

**Discussion by jtrungle** [writer/writer-framework#35] [Ideas]
Currently going into ui.json and trying to match up the ids by clicking around to find usages of an element of the state.

Would be good to have a functionality to search for component id in the builder/ some sort of view to see all the components that are using a particular state element.

**Comment by ramedina86** [comment]
This is a great suggestion. I think a standard search within components would work fine for that purpose.

As a next step, I'm planning a "State Explorer" that gives you a snapshot/dump of state at any given moment. I could add a button like "Check where this state element is used".

**Reply by jtrungle** [reply]
Yes that definitely would help workflow during dev
